functional_tests/steps/bundles.py

```diff
@@ -121,3 +126,166 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
 def bundle_inspect_show(context: Context) -> None:
     expect(context.page.get_by_text("Created by")).to_be_visible()
     expect(context.page.get_by_text(context.bundle_creator.get_username())).not_to_be_visible()
+
+
+# Bundles UI Data Setup
+@given("there are {no_bundles} bundles")
+def multiple_bundles_create(context: Context, no_bundles: str) -> None:
+    context.bundles = []
+    if no_bundles.isdigit():
+        for __ in range(int(no_bundles)):
+            context.bundles.append(BundleFactory(approved=False, created_by=context.user_data["user"]))
+
+
+@given("there are {no_preview_teams} Preview teams")
+def multiple_preview_teams_create(context: Context, no_preview_teams: str) -> None:
+    context.teams = []
+    if no_preview_teams.isdigit():
+        for preview_team_index in range(int(no_preview_teams)):
+            context.teams.append(
+                Team.objects.create(
+                    identifier="preview-team-" + str(preview_team_index), name="Preview_Team_" + str(preview_team_index)
+                )
+            )
+
+
+@given("there are {no_release_calendar} release calendar pages")
+def multiple_release_calendar(context: Context, no_release_calendar: str) -> None:
+    context.release_calendar_pages = []
+    if no_release_calendar.isdigit():
+        for release_calendar_index in range(int(no_release_calendar)):
+            nowish = timezone.now() + timedelta(minutes=5 * (release_calendar_index + 1))
+            context.release_calendar_pages.append(
+                ReleaseCalendarPageFactory(
+                    release_date=nowish,
+                    title="Release Calendar Page" + str(release_calendar_index),
```

Comment:
Similar to above, f-string a better (and I think there's a missing space):

```suggestion
                    title=f"Release Calendar Page {release_calendar_index}",
```

Same for line 174.